thoth/report_processing/pretty_report.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+# thoth-report-processing
+# Copyright(C) 2020 Francesco Murdaca, Sai Sankar Gochhayat
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Utils Create Markdown output from adviser report input."""
+
+import pandas as pd
+from typing import Dict, Any
+
+
+def create_pretty_report_from_json(report: Dict[Any, Any], is_justification: bool = False) -> str:
+    """Create Markdown output from adviser report input."""
+    md = ""
+    if not report:
+        return md
+
+    products = report.get("products")
+    if not products:
+        return md
+
+    md = "Report"
+
+    md += "\n\n" + "Justifications"
+
+    final_df = pd.DataFrame(columns=["message", "type"])
+
+    counter = 0
+    # Consider the first product
+    product = products[0]
+
+    if "justification" in product.keys():
+        justifications = product["justification"]
+
+        if justifications:
+
+            for justification in justifications:
+                final_df.loc[counter] = pd.DataFrame([justification]).iloc[0]
+                counter += 1
+
+    md += "\n\n" + final_df.to_markdown()
+
+    if is_justification:
+        return md
+
+    # Packages in Advised Pipfile
+    md += _add_packages_in_advised_pipfile_to_md_report(product=product, is_dev=False)
+
+    # Dev-Packages in Advised Pipfile
+    md += _add_packages_in_advised_pipfile_to_md_report(product=product, is_dev=True)
+
+    requirements = product["project"]["requirements"]
+
+    if "requires" in requirements:
+        if requirements["requires"]:
+            md += "\n\n" + "Requires in Advised Pipfile"
+            df = pd.DataFrame([requirements["requires"]])
+            md += "\n\n" + df.to_markdown()
+
+    if "source" in requirements:
+        if requirements["source"]:
+            md += "\n\n" + "Source in Advised Pipfile"
+            df = pd.DataFrame(requirements["source"])
```

Comment:
```suggestion
            df = pd.DataFrame([requirements["source"]])
```

Reply:
I think the value of requirements["source"] is an arrary. For example below .. 
line 124 in https://gist.github.com/saisankargochhayat/32fc67949427d200ce5fabe4e15036df